src/features/profile/components/NonBeneficiaryHeader.tsx

```diff
@@ -55,7 +61,21 @@ function NonBeneficiaryHeaderComponent(props: PropsWithChildren<NonBeneficiaryHe
   if (!eligibilityStartDatetime || !eligibilityEndDatetime || today >= eligibilityEndDatetime) {
     body = <BodyContainer testID="body-container-above-18" padding={1} />
   } else if (today >= eligibilityStartDatetime) {
-    if (props.isEligibleForBeneficiaryUpgrade && !props.subscriptionMessage) {
+    if (
+      subscription?.hasIdentityCheckPending &&
+      identification.processing &&
+      !props.subscriptionMessage
+    ) {
+      body = (
+        <BodyContainer testID="body-container-identity-check-pending">
+          <IdentityCheckPendingBadge />
+        </BodyContainer>
+      )
+    } else if (
+      props.isEligibleForBeneficiaryUpgrade &&
+      !props.subscriptionMessage &&
```

Comment:
Je verrai bien un premier `if (props.subscriptionMessage)` pour afficher le message in app, et ensuite tu fais les if else avec les conditions ci-dessus sans le `if !props.subscriptionMessage`